### ISSUE TITLE
All staking rewards are gone after a few blocks

### ISSUE THREAD

**Issue by dragosroua** [gridcoin-community/Gridcoin-Research#758]
I have a loaded client, synced and a valid CPID. Yet all my stake rewards are being rejected. Wallet version: 3.6.3.0-unk. 
`execute beaconstatus`:

```
  | [{"Command" : "beaconstatus"},
{"CPID" : "81bc77d08e3f31050c095d7bbe597b93",
"Beacon Exists" : "Yes",
"Beacon Timestamp" : "10-10-2017 23:06:21",
"Public Key" : "04e61a13c65104acb3035d19ee11aaf49159bf9235e9a845e6b30acc2c6b2608c54f108e5f4edcb72e05bc8979b0f28c6525f9323963ecae4258c1059d0812819e",
"Private Key" : "not shown here",
"Local Configuration Public Key" : "04e61a13c65104acb3035d19ee11aaf49159bf9235e9a845e6b30acc2c6b2608c54f108e5f4edcb72e05bc8979b0f28c6525f9323963ecae4258c1059d0812819e",
"Magnitude (As of last superblock)" : 0.00000000,
"Warning" : "Your magnitude is 0 as of the last superblock: this may keep you from staking POR blocks.",
"Block Signing Test Results" : true,"Configuration Status" : "SUCCESSFUL"}]
-- | --
```
Wallet shows "Boinc mining", I joined two projects (Gridcoin team) for which I'm solving tasks via Boinc (Asteroids@home and Einstein@home). 

From the looks of it, the zero magnitude should be responsible.

Any pointers?

Thank you.

**Comment by tomasbrod** [comment]
How do you know that all your stake rewards are being rejected? How often this happens? Please, try to save a log when this happens and upload it here. Nowadays it is possible to stake with zero magnitude, but it is not recommended as it reverts you into "newbie" state. Invalid beacon could actually prevent staking, but yours is valid. It takes a while for the magnitude to rise to final value. You should verify that you are not mining on a fork, use the gridcoinstats.eu website to compare hashes. Blocks may disappear if you mined them on a fork and then client reorganizes away from the fork, or because the blocks were invalid, or more probably, just randomly (1/2 reorg due to simultaneous stake, happens quite often).

**Comment by dragosroua** [comment]
Thanks for the quick answer, appreciate it!

I know all my stake rewards are rejected because I get an alert in my wallet telling that a new mining  rewards, a few dollars, was generated. I see the transaction in the transaction log for a few dozens of seconds and then it simply disappears. Sometimes it happens immediately, sometimes after a few dozens of seconds.

I compared a recent hash in which I sent from an exchange a few GRC to my wallet and the results are a bit inconsistent.

194adea9f478a0ed3d30034bd1e7d6e361b24fa09e7122fe934449e6b5ad9923

It appears on two blocks, one from 2014 and one from 2017. I can locate the transaction and the address correctly.

What can I do if I'm mining on a fork?

Thank you.

**Comment by tomasbrod** [comment]
To aid in diagnosing this issue and help develop the wallet, please upload the logs right after the next occurrence of this incident. **Please**. The logs are in `debug.log` file in the data directory, same as config. You can drop the file right here into the reply box.

The hash you posted is a block hash (telling me, you could have saved me time searching for it). This block is fairly recent (from yesterday) and appears to be on the main chain, which means you are not on a fork. Why Gridcoinst provided three results? If you look  closely, they are not what you asked for (the hash is different), the search function returned more results, matching various other fields, to stay on the safe side, but they are irrelevant here.

You correctly identified that your stakes are being rejected, I just had to make sure. Now we need to identify why. Considering the fact that the transaction appeared on list, it was accepted locally, but rejected by network.

Further please check:
1. your beacon and dpor configuration
2. version of the wallet (3.6.3.0)

**Comment by denravonska** [comment]
Are the staking transactions still gone now? It could've been a display bug.

**Comment by denravonska** [comment]
This will most likely be solved with #774.
